### DISCUSSION TITLE
Please, remove the delay after next video playlist button

### DISCUSSION THREAD

**Discussion by baack** [smplayer-dev/smplayer#122] [General]
Please, remove the delay after next video playlist button. It was added as workaround to avoid crash of mpv when going next video reasonably fast.

I was one who provided a bug report of smplayer showing mpv crash dialog if Play Next is pressed too fast. On the old forum.
You said that you could not find a proper fix yet but added (r9467) a workaround for the play next problem.

Now **I beg you to remove this workaround** (an awful delay) and fix the actual issue (crash on fast video change).
The reasons are:
1) The delay is super-disturbing. I used to scroll though the playlist with next-next button and getting video frames rather fast. It always had worked almost like scrolling in image viewer! And it was awesome and nice feature of smplayer as a fast and solid player. No I can't do that anymore.
2) Scrolling videos like that had no issues for ages. You can find the commit that added this bug and analyze it.
3) The actual issue is not fixed at all. You still can get crash of mpv just by clicking in playlist with mouse too fast. So, the workaround is ugly, not covering all cases and does not solve the actual problem (crash is a problem!)

But smplayer is great and I really hope the delay and crashes will be gone. I beg you to do that! :)

**Comment by baack** [comment]
Update. Just tried to crash mpv from smplayer with fast change of current file with mouse clicks and it never crashed.
Maybe the issue with crash is fixed in MPV 0.33.1!

In this case it makes even more sense to remove the annoying delay of play-next play-previous playlist functions. Should I create a separate feature request/issue for this workaround to be removed? Thanks in advance.

**Comment by smplayer-dev** [comment]
What version are you using? I think that delay was removed some time ago.

**Reply by baack** [reply]
21.1.0 (revision 9475)

**Reply by smplayer-dev** [reply]
Try [version 9724](https://github.com/smplayer-dev/smplayer/releases/tag/v21.1.0.9724).

**Comment by GeneBenson** [comment]
This version solves the issue of the delay on both Linux Mint and Debian (tracking testing). I hope it also works for baack.

**Comment by baack** [comment]
I checked:

> Version: 21.1.0 (revision 9724) Development version
> Using Qt 5.15.2 (compiled with Qt 5.12.7)
> Using MPV 0.33.1

And face 2 issues 
- the delay still exists, but is smaller.
- mpv crashes on fast switching.

P.S. I used opensuse rpm version to install it on Fedora 33 if it matters.

**Comment by baack** [comment]
I checked with MPlayer (version SVN r38132) and MPlayer does not crash.

About delay. To reproduce:
1) Add 10 files to the playlist
2) Start first
3) Start pressing Next fast (using keyboard shortcut or playlist button with mouse)

What happens: The playing video is the same, the cursor is moving along playlist.

P.S. After the first switch the pause is not noticeable.

**Comment by AH12A** [comment]
It would be really useful if we could get this fix released as 21.1.1 form so package managers would pick it up
